x-pack/plugins/apm/server/routes/transactions.ts

```diff
@@ -200,23 +206,42 @@ export const transactionLatencyChatsRoute = createRoute({
       logger,
     };
 
-    const [latencyData, anomalyTimeseries] = await Promise.all([
+    const [
+      currentPeriod,
+      anomalyTimeseries,
+      previousPeriod,
+    ] = await Promise.all([
       getLatencyTimeseries({
         ...options,
+        start,
+        end,
         latencyAggregationType: latencyAggregationType as LatencyAggregationType,
       }),
       getAnomalySeries(options).catch((error) => {
         logger.warn(`Unable to retrieve anomalies for latency charts.`);
         logger.error(error);
         return undefined;
       }),
+      comparisonStart && comparisonEnd
+        ? getLatencyTimeseries({
+            ...options,
+            start: comparisonStart,
+            end: comparisonEnd,
+            latencyAggregationType: latencyAggregationType as LatencyAggregationType,
+          }).then((latency) => ({
+            ...latency,
+            latencyTimeseries: offsetPreviousPeriodCoordinates({
+              currentPeriodStart: start,
+              previousPeriodStart: comparisonStart,
+              previousPeriodTimeseries: latency.latencyTimeseries,
+            }),
+          }))
+        : { latencyTimeseries: [], overallAvgDuration: null },
```

Comment:
This is a little hard to read IMO. It might deserve a separate function. Maybe both current and previous period can be returned by one function.